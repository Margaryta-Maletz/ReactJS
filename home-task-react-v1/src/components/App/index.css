.wrapper {
    width: 1200px;
    box-sizing: border-box;
    padding: 20px 60px;
    display: flex;
}
