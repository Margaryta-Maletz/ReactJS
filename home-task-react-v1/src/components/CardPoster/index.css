.card-poster_wrapper {
    width: 322px;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;

    font-family: Montserrat;
    font-style: normal;
    font-weight: 500;
    color: white;
    mix-blend-mode: normal;
}

.card-poster_title {
    font-size: 18px;
    line-height: 22px;
    opacity: 0.7;
    margin: 22px 0 8px;
    width: 75%;
}

.card-poster_genre {
    font-size: 14px;
    line-height: 17px;
    opacity: 0.5;
    margin-block: 0;
    margin: 0 0 50px;
}

.card-poster_release-date {
    width: 66px;
    height: 26px;
    border: 1px solid #555555;
    box-sizing: border-box;
    border-radius: 4px;
    font-size: 14px;
    line-height: 24px;
    text-align: center;
    margin: 22px 0 8px;
}
