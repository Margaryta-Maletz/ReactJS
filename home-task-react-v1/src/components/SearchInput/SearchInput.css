.search-input_wrapper {
    margin: 0 auto;
    height: 145px;
}

.search-input_header {
    font-family: Montserrat;
    font-style: normal;
    font-weight: 300;
    font-size: 40px;
    line-height: 49px;
    text-align: left;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: white;
    margin: 0 0 38px;
}

.search-input_input {
    width: 673px;
    height: 57px;
    mix-blend-mode: normal;
    opacity: 70%;
    background-color: rgba(50, 50, 50, 0.8);
    font-style: normal;
    font-weight: normal;
    font-size: 20px;
    line-height: 24px;
    text-align: left;
    padding: 0 20px;
    vertical-align: top;
    color: white;
    margin-right: 14px;
    border: none;
}

.search-input_button {
    width: 233px;
    height: 57px;
    background: #F65261;
    border-radius: 4px;
    font-style: normal;
    font-weight: 500;
    font-size: 20px;
    line-height: 24px;
    text-align: center;
    text-transform: uppercase;
    color: white;
}
