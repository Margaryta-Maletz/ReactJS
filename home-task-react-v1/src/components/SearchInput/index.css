.search-input_wrapper {
    vertical-align: center;
    margin-bottom: 10px;
}
.search-input_header {
    font-family: Montserrat;
    font-style: normal;
    font-weight: 300;
    font-size: 40px;
    line-height: 49px;
    text-align: left;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: white;
}

.search-input_input {
    width: 713px;
    height: 57px;
    mix-blend-mode: normal;
    opacity: 70%;
    background-color: rgba(50, 50, 50, 0.8);
    font-style: normal;
    font-weight: normal;
    font-size: 20px;
    line-height: 24px;
    text-align: left;
    vertical-align: top;
    mix-blend-mode: normal;
    opacity: 30%;
    color: white;
    margin-right: 14px;
}

.search-input_button {
    width: 233px;
    height: 57px;
    background: #F65261;
    border-radius: 4px;
    font-style: normal;
    font-weight: 500;
    font-size: 20px;
    line-height: 24px;
    text-align: center;
    text-transform: uppercase;
    color: white;
}
