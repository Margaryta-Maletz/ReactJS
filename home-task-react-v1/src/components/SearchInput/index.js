import React from 'react';
import './index.css';

export const SearchInput = () => {
    return (
        <div className="search-input_wrapper">
            <h3 className='search-input_header'>Find your movie</h3>
            <input className='search-input_input' placeholder='What do you want to watch?'></input>
            <button className='search-input_button'>search</button>
        </div>
    )
}
