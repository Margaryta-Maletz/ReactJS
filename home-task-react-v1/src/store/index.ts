import thunk from 'redux-thunk';
import {configureStore, getDefaultMiddleware, ThunkAction, Action} from '@reduxjs/toolkit';
import rootReducer from './slice';

const middleware = getDefaultMiddleware({
    imitableCheck: false,
    serializableCheck: false,
    thunk: true,
});

export const store = configureStore({
    reducer: rootReducer,
    middleware: middleware,
    devTools: process.env.NODE_ENV !== 'production',
});

/*export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> =
    ThunkAction<
        ReturnType,
        RootState,
        unknown,
        Action<string>
    >;*/
