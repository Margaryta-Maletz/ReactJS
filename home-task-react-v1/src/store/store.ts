import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import rootReducer from './reducers';

export const store = createStore(
    rootReducer,
    applyMiddleware(thunk)
);

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> =
    ThunkAction<
        ReturnType,
        RootState,
        unknown,
        Action<string>
    >;
