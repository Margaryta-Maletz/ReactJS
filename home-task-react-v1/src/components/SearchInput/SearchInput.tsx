import React, {useEffect} from 'react';
import './SearchInput.css';
import {useHistory} from "react-router-dom";
import {useDispatch, useSelector} from "react-redux";
import {IState} from "../../store/types";
import {setSearchString, setEditMovieList} from '../../store/slice';

export const SearchInput: React.FC<{searchQuery: string}> = (props) => {
    const history = useHistory();
    const dispatch = useDispatch();
    const { searchString } = useSelector((state: IState) => state);
    const handleChangeSearchString = (e: React.ChangeEvent<HTMLInputElement>) => {
        dispatch(setSearchString(e.target.value));
    }

    useEffect(() => {
        const searchQuery = props.searchQuery || "";
        dispatch(setSearchString(searchQuery)) && dispatch(setEditMovieList(true));
    }, []);

    const handleOnClick = () => {
        history.push(`/search/${searchString}`);
        dispatch(setEditMovieList(true));
    }

    return (
        <div className="search-input_wrapper">
            <h3 className='search-input_header'>Find your movie</h3>
            <input className='search-input_input' placeholder='What do you want to watch?' value={searchString} onChange={handleChangeSearchString} onKeyDown={(e) => (e.code === 'Enter' && handleOnClick())   }/>
            <button className='search-input_button' onClick={handleOnClick}> search </button>
        </div>
    )
}
