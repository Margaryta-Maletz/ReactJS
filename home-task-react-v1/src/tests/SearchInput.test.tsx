import React from "react";
import { render } from '@testing-library/react';
import { SearchInput } from '../components/SearchInput/';
import {Provider} from "react-redux";
import {store} from "../store/store";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
        pathname: "localhost:8080/search/"
    }),
    useHistory: (path: string) => ({
        push: jest.fn(),
        location: {
            pathname: "localhost:8080/search/" + path
        }
    })
}));

describe('make snapshot', () => {
    test('renders', () => {
        const { container } = render(
            <Provider store={store}>
                <SearchInput />
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
});
