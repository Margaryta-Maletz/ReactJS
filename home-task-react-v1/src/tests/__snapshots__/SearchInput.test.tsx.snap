// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`make snapshot renders 1`] = `
<div>
  <div
    class="search-input_wrapper"
  >
    <h3
      class="search-input_header"
    >
      Find your movie
    </h3>
    <input
      class="search-input_input"
      placeholder="What do you want to watch?"
      value=""
    />
    <button
      class="search-input_button"
    >
       search 
    </button>
  </div>
</div>
`;
